### PR TITLE
rtr_mgr: wrap tommy_list in rtr_mgr_config in a struct

### PR DESCRIPTION
**Pull request by mroethke**
The purpose of this is to hide tommy symbols from public headers without
forward declaring tommy_list. Which is only possible with C11 as it is
typedefed.

#184

### CODE REVIEW

File: rtrlib/rtr_mgr.h [hunk 1]
```diff
@@ -75,9 +74,11 @@ typedef void (*rtr_mgr_status_fp)(const struct rtr_mgr_group *,
 				  const struct rtr_socket *,
 				  void *);
 
+struct tommy_list_wrapper;
+
 //TODO Add refresh, expire, and retry intervals to config for easier access.
 struct rtr_mgr_config {
-	tommy_list groups;
+	struct tommy_list_wrapper *groups;
```
Comment (smlng) [inline rtrlib/rtr_mgr.h]: does this need to be a pointer?

Comment (mroethke) [reply rtrlib/rtr_mgr.h]: Yes, because tommy_list_wrapper due to the forward declaration the size of tommy_list_wrapper in rtr_mgr.h.

Comment (smlng) [reply rtrlib/rtr_mgr.h]: thanks, understood

**Comment by mroethke** [comment]
ping @waehlisch

**Pull request merged**